{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from subprocess import Popen, PIPE, call\n",
    "path_target_finder = '/home/juan/Desktop/juan/bio/mrcv/sw/TargetFinder/targetfinder.pl'\n",
    "target_seqs = '../../data/res/DEG-cdna.fasta'\n",
    "file_deg_mirna_fasta = '../../data/res/DEG-miRNA.fasta'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Locus</th>\n",
       "      <th>Name</th>\n",
       "      <th>Length</th>\n",
       "      <th>Reads</th>\n",
       "      <th>RPM</th>\n",
       "      <th>UniqueReads</th>\n",
       "      <th>FracTop</th>\n",
       "      <th>Strand</th>\n",
       "      <th>MajorRNA</th>\n",
       "      <th>MajorRNAReads</th>\n",
       "      <th>...</th>\n",
       "      <th>PhaseScore</th>\n",
       "      <th>Short</th>\n",
       "      <th>Long</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "      <th>23</th>\n",
       "      <th>24</th>\n",
       "      <th>Gene</th>\n",
       "      <th>log2FoldChange</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2A:652328872-652329011</td>\n",
       "      <td>Cluster_6291</td>\n",
       "      <td>140</td>\n",
       "      <td>1185</td>\n",
       "      <td>9.300</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001</td>\n",
       "      <td>-</td>\n",
       "      <td>UGCCAAAGGAGAGUUGCCCUG</td>\n",
       "      <td>1136</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1155</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>Cluster_6291</td>\n",
       "      <td>-1.193250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2A:707747484-707747579</td>\n",
       "      <td>Cluster_6500</td>\n",
       "      <td>96</td>\n",
       "      <td>1395</td>\n",
       "      <td>10.948</td>\n",
       "      <td>1</td>\n",
       "      <td>0.999</td>\n",
       "      <td>+</td>\n",
       "      <td>UGGACGAGGAUGUGCAGCUGC</td>\n",
       "      <td>1119</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1385</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Cluster_6500</td>\n",
       "      <td>0.832313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2D:151985102-151985210</td>\n",
       "      <td>Cluster_9467</td>\n",
       "      <td>109</td>\n",
       "      <td>3607</td>\n",
       "      <td>28.309</td>\n",
       "      <td>59</td>\n",
       "      <td>1.000</td>\n",
       "      <td>+</td>\n",
       "      <td>AUCAGGAGAGAUGACACCGAC</td>\n",
       "      <td>2753</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>93</td>\n",
       "      <td>3</td>\n",
       "      <td>312</td>\n",
       "      <td>3014</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>161</td>\n",
       "      <td>Cluster_9467</td>\n",
       "      <td>1.259891</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   #Locus          Name  Length  Reads     RPM  UniqueReads  \\\n",
       "0  2A:652328872-652329011  Cluster_6291     140   1185   9.300            1   \n",
       "1  2A:707747484-707747579  Cluster_6500      96   1395  10.948            1   \n",
       "2  2D:151985102-151985210  Cluster_9467     109   3607  28.309           59   \n",
       "\n",
       "   FracTop Strand               MajorRNA  MajorRNAReads       ...        \\\n",
       "0    0.001      -  UGCCAAAGGAGAGUUGCCCUG           1136       ...         \n",
       "1    0.999      +  UGGACGAGGAUGUGCAGCUGC           1119       ...         \n",
       "2    1.000      +  AUCAGGAGAGAUGACACCGAC           2753       ...         \n",
       "\n",
       "   PhaseScore  Short Long   20    21  22  23   24          Gene  \\\n",
       "0        -1.0     14    0    3  1155  11   2    0  Cluster_6291   \n",
       "1        -1.0      7    2    0  1385   1   0    0  Cluster_6500   \n",
       "2        -1.0     93    3  312  3014   9  15  161  Cluster_9467   \n",
       "\n",
       "   log2FoldChange  \n",
       "0       -1.193250  \n",
       "1        0.832313  \n",
       "2        1.259891  \n",
       "\n",
       "[3 rows x 23 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('../../data/ss/04_02_19/DEG.Y.csv', sep='\\t')\n",
    "print(len(df.index))\n",
    "df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records = []\n",
    "for k,v in df.iterrows():\n",
    "    record = SeqRecord(Seq(v.MajorRNA), id=v.Name, description=str(v.log2FoldChange))\n",
    "    records.append(record)\n",
    "SeqIO.write(records, file_deg_mirna_fasta, 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12\n",
      "0\n",
      "b'No results for Cluster_6291\\n' b''\n",
      "1\n",
      "b'Cluster_6500\\tTraesCS6D02G296600.1 cdna chromosome:IWGSC:6D:408147367:408147870:1 gene:TraesCS6D02G296600 ge\\t28\\t47\\t+\\t3.5\\tGCAGC-GUGCAUCCUCGUCCU\\t::::: :..::::::::::: \\tCGUCGACGUGUAGGAGCAGGU\\nCluster_6500\\tTraesCS7A02G231200.1 cdna chromosome:IWGSC:7A:201985643:201990646:-1 gene:TraesCS7A02G231200 g\\t778\\t798\\t+\\t4\\tGCAGCUGCAGAUCCUCGUCGA\\t::::::::: ::::::::: :\\tCGUCGACGUGUAGGAGCAGGU\\n' b''\n",
      "2\n",
      "b'No results for Cluster_9467\\n' b''\n",
      "3\n",
      "b'Cluster_9839\\tTraesCS2B02G490400.1 cdna chromosome:IWGSC:2B:688234797:688235685:-1 gene:TraesCS2B02G490400 g\\t264\\t283\\t+\\t3.5\\tGCUGGCCAAGGUGGCCGCCG\\t:::.::::::::..::::: \\tCGAUCGGUUCCAUUGGCGGA\\n' b''\n",
      "4\n",
      "b'Cluster_10532\\tTraesCS6D02G296600.1 cdna chromosome:IWGSC:6D:408147367:408147870:1 gene:TraesCS6D02G296600 ge\\t28\\t47\\t+\\t3.5\\tGCAGC-GUGCAUCCUCGUCCU\\t::::: :..::::::::::: \\tCGUCGACGUGUAGGAGCAGGU\\nCluster_10532\\tTraesCS7A02G231200.1 cdna chromosome:IWGSC:7A:201985643:201990646:-1 gene:TraesCS7A02G231200 g\\t778\\t798\\t+\\t4\\tGCAGCUGCAGAUCCUCGUCGA\\t::::::::: ::::::::: :\\tCGUCGACGUGUAGGAGCAGGU\\n' b''\n",
      "5\n",
      "b'No results for Cluster_17194\\n' b''\n",
      "6\n",
      "b'No results for Cluster_17678\\n' b''\n",
      "7\n",
      "b'Cluster_18555\\tTraesCS4A02G075600.1 cdna chromosome:IWGSC:4A:75213035:75215412:-1 gene:TraesCS4A02G075600 gen\\t1558\\t1577\\t+\\t4\\tGCUCGCCAAGAAGACCAACG\\t::: :::::::::::: :: \\tCGAUCGGUUCUUCUGGGUGG\\n' b''\n",
      "8\n",
      "b'No results for Cluster_19565\\n' b''\n",
      "9\n",
      "b'No results for Cluster_30749\\n' b''\n",
      "10\n",
      "b'No results for Cluster_32170\\n' b''\n",
      "11\n",
      "b'No results for Cluster_33754\\n' b''\n"
     ]
    }
   ],
   "source": [
    "#search targets\n",
    "out_file = open(\"../../data/res/04_02_19/targets_cdna.csv\",\"w\")\n",
    "print(len(df.index))\n",
    "out_total = ''\n",
    "for k,v in df.iterrows():\n",
    "    print(k,)\n",
    "    if 'N' in v.MajorRNA:\n",
    "        print(v.Name + \"has Ns\")\n",
    "        continue\n",
    "    cmd_list = ['perl', path_target_finder,'-s',v.MajorRNA,'-q',v.Name,'-d',target_seqs,'-p','table']\n",
    "#    print(' '.join(cmd_list))\n",
    "    pro = Popen(cmd_list, stdout=PIPE, stderr=PIPE)\n",
    "    out,err = pro.communicate()\n",
    "#    out = str(out)\n",
    "#    err = str(err)\n",
    "    print(out,err)\n",
    "    if b\"No results for\" in out:\n",
    "        continue\n",
    "#    out_total += out.decode(\"utf-8\") \n",
    "    out_file.write(out.decode(\"utf-8\") )\n",
    "out_file.close()\n",
    "#os.system('grep -v \"No results for\" targets.csv > targets.clean.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
